### ISSUE TITLE
Constant diff on `tfe_registry_module` with v0.51.1

### ISSUE THREAD

**Issue by jamesmbourne** [hashicorp/terraform-provider-tfe#1216]
#### Terraform Cloud/Enterprise version

Terraform Cloud

#### Terraform version

```plaintext
1.6.6
```

#### Terraform Configuration Files

```terraform
resource "tfe_registry_module" "module" {
  vcs_repo {
    oauth_token_id     = data.tfe_oauth_client.client.oauth_token_id
    identifier         = github_repository.module.full_name
    display_identifier = github_repository.module.full_name
  }
}
```

#### Debug Output

```plaintext
...
```

#### Expected Behavior

No diff when resource is up-to-date

#### Actual Behavior

Resources were stuck in a cycle of being destroyed and created again because of a diff in the `vcs_repo` block.

<img width="1607" alt="Screenshot 2024-01-16 at 13 34 30" src="https://github.com/hashicorp/terraform-provider-tfe/assets/594697/99a08939-ceaf-466c-825a-9c0b5b543bbb">

#### Additional Context

We are using tag-based publishing for modules.

I also tried using the config
```hcl
resource "tfe_registry_module" "module" {
  vcs_repo {
    tags               = true
    branch             = ""
    oauth_token_id     = data.tfe_oauth_client.client.oauth_token_id
    identifier         = github_repository.module.full_name
    display_identifier = github_repository.module.full_name
  }
}
```
but this continued to show a diff and try to re-create the resources.

It appears this was introduced by https://github.com/hashicorp/terraform-provider-tfe/pull/1096. Downgrading to `v0.50.0` resolves the issue.

**Comment by brianpham** [comment]
We are seeing the same issue for our org. Downgrading to `v0.50.0` to test now.

**Comment by brianpham** [comment]
Downgrading to `v0.50.0` works for us

**Comment by LiamLogan1996** [comment]
Hi @brianpham We were able to fix this by adding the organisation argument to the tfe_registry_module resource block. 

As it states in the docs, if you have module_providev specified or vcs_repo is used via a GitHub App you must have the organisation argument set.

Hope this helps

**Comment by iniinikoski** [comment]
Hi! Great to "bump" into this issue here! :) 

Noticed also this changed behaviour ; I had to do two things (using the latest provider version 0.52.0):

1) added the TFE_ORGANIZATION variable (or, set the organization in the provider config)
2) as we use tag-based publishing, first-time ever, I had to set this in our custom module (I think it's been a default to true earlier, looks like it at least)

And, we're not (still yet) using a GitHub app for GitHub access but the "old-school" OAuth. Hmm. Interesting.
